### PR TITLE
Document minutely probes

### PR DESCRIPTION
**Pull request by tombruijn**
Part of https://github.com/appsignal/appsignal-server/issues/3807

Merge when this feature has been shipped in Ruby gem 2.9.0.

### CODE REVIEW

File: source/ruby/instrumentation/minutely-probes.html.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
+---
+title: "Minutely probes"
+---
+
+Minutely probes are a method to periodically send custom metrics to AppSignal. This is a system that is included in the AppSignal Ruby gem by default. At the start of every minute the minutely probes are triggered on by one to collect metrics and then snoozed until the next minute.
```
Comment (thijsc) [inline source/ruby/instrumentation/minutely-probes.html.md]:
```suggestion
Minutely probes are a mechanism to periodically send custom metrics to AppSignal. This is a system that is included in the AppSignal Ruby gem by default. At the start of every minute the minutely probes are triggered on by one to collect metrics and then snoozed until the next minute.
```

File: data/config_options.yml [hunk 2]
```diff
@@ -203,6 +203,16 @@ options:
       default_value: true
     description: |
       Set this option to `false` to disable [host metrics](/metrics/host.html) collection.
+  - config_key: enable_minutely_probes
+    env_key: APPSIGNAL_ENABLE_MINUTELY_PROBES
+    ruby:
+      since: 1.3.0
+      since_notes:
+        - "`2.9.0`: Default value was change to `true`."
```
Comment (jvanbaarsen) [inline data/config_options.yml]:
```suggestion
        - "`2.9.0`: Default value was changed to `true`."
```

File: source/ruby/instrumentation/minutely-probes.html.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
+---
+title: "Minutely probes"
+---
+
+Minutely probes are a mechanism to periodically send custom metrics to AppSignal. This is a system that is included in the AppSignal Ruby gem by default. At the start of every minute the minutely probes are triggered on by one to collect metrics and then snoozed until the next minute.
```
Comment (jeffkreeftmeijer) [inline source/ruby/instrumentation/minutely-probes.html.md]:
```suggestion
Minutely probes are a mechanism to periodically send custom metrics to AppSignal. This is a system that is included in the AppSignal Ruby gem by default. At the start of every minute the minutely probes are triggered one by one to collect metrics and then snoozed until the next minute.
```

File: source/ruby/instrumentation/minutely-probes.html.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  - [Lambda probe](#lambda-probe)
+  - [Class probe](#class-probe)
+  - [Initialized class probe](#initialized-class-probe)
+- [Registering probes](#registering-probes)
+  - [Deprecated registration method](#deprecated-registration-method)
+- [Overriding default probes](#overriding-default-probes)
+
+## Usage
+
+The minutely probes allows the AppSignal Ruby gem to collect [custom metrics](/metrics/custom.html) by default for [integrations](/ruby/integrations) and app specific metrics by [creating your own probe](#creating-probes).
```
Comment (jeffkreeftmeijer) [inline source/ruby/instrumentation/minutely-probes.html.md]:
```suggestion
The minutely probes allow the AppSignal Ruby gem to collect [custom metrics](/metrics/custom.html) by default for [integrations](/ruby/integrations) and app-specific metrics by [creating your own probe](#creating-probes).
```

File: source/ruby/instrumentation/minutely-probes.html.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+### Multiple instances
+
+Once activated the minutely probes system runs on every instance of an app. This means that if a probe report metrics without some kind of differentiating tag, global metrics may be overwritten by instance-level metrics. For example, there's a probe that tracks how large the local background job queue is for an app. The queue database runs locally on the instance and queue sizes may vary wildly. Each instances of an app report different metrics values for the same metric and tags, overwriting the metric value every minute with those of the last reporting instance.
```
Comment (jeffkreeftmeijer) [inline source/ruby/instrumentation/minutely-probes.html.md]:
```suggestion
Once activated the minutely probes system runs on every instance of an app. This means that if a probe report metrics without some kind of differentiating tag, global metrics may be overwritten by instance-level metrics. For example, there's a probe that tracks how large the local background job queue is for an app. The queue database runs locally on the instance and queue sizes may vary wildly. Each instance of an app reports different metric values for the same metric and tags, overwriting the metric value every minute with those of the last reporting instance.
```

File: source/ruby/instrumentation/minutely-probes.html.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+To remedy this, we suggest [tagging](/metrics/custom.html#metric-tags) your metrics with the hostname or something else unique for each instance. For example, the [Sidekiq probe](/ruby/integrations/sidekiq.html#minutely-probe) tags metrics by default with the Redis hostname.
```
Comment (jeffkreeftmeijer) [inline source/ruby/instrumentation/minutely-probes.html.md]:
```suggestion
To remedy this, we suggest [tagging](/metrics/custom.html#metric-tags) your metrics with the hostname or something else unique for each instance. For example, the [Sidekiq probe](/ruby/integrations/sidekiq.html#minutely-probe) tags metrics with the Redis hostname by default.
```

File: source/ruby/instrumentation/minutely-probes.html.md [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+
+### Rails apps
+
+AppSignal minutely probes need to be registered while your Rails app boots. We recommend registering probes in an AppSignal initializer: `config/initializers/appsignal.rb`.
+
+### Ruby apps
+
+Register you probe in a file that's called on boot before `Appsignal.start` is called. For pure-Ruby/non-Rails apps probes needs to be registered in a file that's loaded on boot.
+
+See also [Integration AppSignal](/ruby/instrumentation/integrating-appsignal.html) for instructions how to integrate AppSignal in your app.
```
Comment (jeffkreeftmeijer) [inline source/ruby/instrumentation/minutely-probes.html.md]:
```suggestion
See also [Integrating AppSignal](/ruby/instrumentation/integrating-appsignal.html) for instructions how to integrate AppSignal in your app.
```

**Comment by tombruijn** [comment]
Waiting until we release the next Ruby gem minor version to publish this.

**Comment by matsimitsu** [comment]
Added MongoDB docs

**Pull request merged**